File: Kirameku-backend/admin/src/api/category.ts
import { http } from "@/utils/http";

export type CategoryItem = {
  id: number;
  name: string;
  slug: string;
  description: string;
  sort: number;
  post_count: number;
  created_at: string;
  updated_at: string;
};

/** 获取分类列表 */
export const getCategories = () => {
  return http.request<CategoryItem[]>("get", "/api/categories");
};

/** 创建分类 */
export const createCategory = (data: {
  name: string;
  slug: string;
  description?: string;
  sort?: number;
}) => {
  return http.request<CategoryItem>("post", "/api/categories", { data });
};

/** 更新分类 */
export const updateCategory = (
  catId: number,
  data: { name?: string; slug?: string; description?: string; sort?: number }
) => {
  return http.request<CategoryItem>("put", `/api/categories/${catId}`, { data });
};

/** 删除分类 */
export const deleteCategory = (catId: number) => {
  return http.request<{ ok: boolean }>("delete", `/api/categories/${catId}`);
};

File: Kirameku-backend/app/schemas/category.py
from datetime import datetime
from pydantic import BaseModel


class CategoryCreate(BaseModel):
    name: str
    slug: str
    description: str = ""
    sort: int = 0


class CategoryUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    description: str | None = None
    sort: int | None = None


class CategoryOut(BaseModel):
    id: int
    name: str
    slug: str
    description: str
    sort: int
    post_count: int
    created_at: datetime
    updated_at: datetime


class TagCreate(BaseModel):
    name: str
    slug: str


class TagUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None


class TagOut(BaseModel):
    id: int
    name: str
    slug: str
    post_count: int
